### ISSUE TITLE
(Warning) shift creation: left < right

### ISSUE THREAD

**Issue by LEARNTAU** [IDNI/tau-lang#15]
Tau version: 0.7 (v0.7-alpha-145-g82e48cd)
Ubuntu Linux OS

Hello, I tried to play with the code provided by ptt kk shown in the Tau dev chat.

# input

succ(int[0](1)) := int[1](1)
succ(int[1](1)) := int[2](1)
succ(int[2](1)) := int[3](1)
succ(int[3](1)) := int[4](1)
succ(int[4](1)) := int[5](1)
pred(int[t](1)) := int[t-1](1)
add(x, int[0](1)) := x
add(int[0](1), x) := x
add(x, y) := add(pred(x), succ(y))
mul(x, int[1](1)) := x
mul(int[1](1), x) := x
mul(x, y) := add(y, mul(pred(x), y))

Everything seems to work properly, but when I entered 

n mul(int[0](1),int[2](1))

I got this non-stop message: (Warning) shift creation: left < right 

![p1](https://github.com/user-attachments/assets/e0d278a4-0c43-4b63-88f6-d2f1e6ffb73b)
![p2](https://github.com/user-attachments/assets/b7fd6127-570b-4826-b63d-f65c20f28fdc)

I cant's stop the message from appearing continuously. Is this a bug?

**Comment by pt7k** [comment]
Good catch. It goes into some sort of endless loop due to an unhandled edge case in the multiplication recurrence.
On tau version (v0.7-alpha-136-g6df5e78) it would just go on forever without any warning messages, but if you "set severity debug" then it spits out the ever looping recurrence and crashes after a while.
Obviously the edge case can be easily fixed for the multiplication example by adding the following lines before the mul(x, y) definition's line:
```
mul(x, int[0](1)) := int[0](1)
mul(int[0](1), x) := int[0](1)
```
but still this example shows that an endless loop can be created while using recurrence functions.

**Comment by castrod** [comment]
This issue was solved long time ago and is not reproducible anymore. Thanks a lot for reporting it.

<img width="1000" height="489" alt="Image" src="https://github.com/user-attachments/assets/2e85be9d-68c5-40ce-baaf-87ebc862afb9" />
